packages/actions/src/releasePackages/generateReleaseTree.ts

```diff
@@ -0,0 +1,125 @@
+import { create } from '@actions/glob';
+
+const nonNodePackages = new Set(['@discordjs/proxy-container']);
+const globber = await create(`packages/*/package.json`);
+
+interface PackageJSON {
+	dependencies?: Record<string, string>;
+	name?: string;
+	private?: boolean;
+	version?: string;
+}
+
+export interface ReleaseEntry {
+	dependsOn?: string[];
+	name: string;
+	version: string;
+}
+
+async function getReleaseEntries() {
```

Comment:
This function could leverage the `pnpm list` command instead of manually finding the packages and their dependencies, e.g.:

```typescript
function getReleaseEntries(): ReleaseEntry[] {
	const pkgList = JSON.parse(execSync("pnpm list --recursive --only-projects --prod --json").toString());
	return pkgList
		.filter((pkg) => !pkg.private && !nonNodePackages.has(pkg.name))
		.map((pkg) => ({
			name: pkg.name,
			version: pkg.version,
			dependsOn: pkg.dependencies ? Object.keys(pkg.dependencies) : [],
		}));
}
```

Reply:
I do like this, but I really don't like that it gives me `@discordjs/discordjs` (the monorepo), and the apps directory. Right now, they're all marked private appropriately, so it'd work just fine, just rubs me  the wrong way a bit.

Reply:
oh! I can still use --filter with this, works great